src/rgw/rgw_op.cc

```diff
@@ -702,26 +702,25 @@ static void rgw_add_grant_to_iam_environment(rgw::IAM::Environment& e, struct re
   }
 }
 
-rgw::IAM::Environment rgw_build_iam_environment(RGWRados* store,
-						struct req_state* s)
+void rgw_build_iam_environment(RGWRados* store,
+	                              struct req_state* s)
 {
-  rgw::IAM::Environment e;
   const auto& m = s->info.env->get_map();
```

Comment:
I'd create a local reference to s->env and use that in the body, I think

Reply:
I have changed this, because before this method is called, iam env is populated in this newly added method: rgw::auth::WebIdentityApplier::modify_request_state(), hence using a local reference overwrites the already existing values.